docs/topics/compatibility-guides/compatibility-guide-21.md

```diff
@@ -0,0 +1,479 @@
+> - 2.1.0: enable the new behavior by default
+
+### Correct handling of private members conflicting with public members in Kotlin/Native
+
+> **Issue**: [KTLC-43](https://youtrack.jetbrains.com/issue/KTLC-43)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: behavioral
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: private functions and properties in Kotlin/Native no longer override or affect public members in a superclass, aligning with JVM behavior
+
+### Forbid access to private operator functions in public inline functions
+
+> **Issue**: [KTLC-71](https://youtrack.jetbrains.com/issue/KTLC-71)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Private operator functions such as `getValue()`, `setValue()`, `provideDelegate()`, `hasNext()`, and `next()` can no longer be accessed in public inline functions
+>
+> **Deprecation cycle**:
+>
+> - 2.0.0: report a deprecation warning for accessing private operator functions in public inline functions
+> - 2.1.0: raise the warning to an error
+
+### Prohibit passing invalid arguments to invariant parameters annotated with @UnsafeVariance
+
+> **Issue**: [KTLC-72](https://youtrack.jetbrains.com/issue/KTLC-72)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The compiler now ignores `@UnsafeVariance` annotations during type checks, enforcing stricter type safety for invariant type parameters.
+> This prevents invalid calls that rely on `@UnsafeVariance` to bypass expected type checks.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: activate the new behavior
+
+### Report nullability errors for error-level nullable arguments of warning-level Java types
+
+> **Issue**: [KTLC-100](https://youtrack.jetbrains.com/issue/KTLC-100)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The compiler now detects nullability mismatches in Java methods where a warning-level
+> nullable type contains type arguments with stricter, error-level nullability.
+> This ensures that previously ignored errors in type arguments are reported correctly.
+>
+> **Deprecation cycle**:
+>
+> - 2.0.0: report a deprecation warning for nullability mismatches in Java methods with stricter type arguments
+> - 2.1.0: raise the warning to an error
+
+### Report implicit usages of inaccessible types
+
+> **Issue**: [KTLC-3](https://youtrack.jetbrains.com/issue/KTLC-3)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The compiler now reports usages of inaccessible types in function literals and type arguments,
+> preventing compilation and runtime failures caused by incomplete type information.
+>
+> **Deprecation cycle**:
+>
+> - 2.0.0: report warnings for function literals with parameters or receivers of inaccessible non-generic types and for types with inaccessible type argument; report errors for
```

Comment:
Would it work if we write something like:
2.0.0: reports both warnings and errors; warnings for function literals with parameters or receivers of inaccessible non-generic types and for types with inaccessible type argument; errors for function literals with parameters or receivers of inaccessible generic types and for types with inaccessible generic type arguments in specific scenarios

Reply:
I like this but I actually prefer explicitly separating warnings and errors into two clauses, I believe it makes it easier to follow